x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.protocol.xpack.XPackUsageRequest;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.application.search.SearchApplicationIndexService;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureAction;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureResponse;
+import org.elasticsearch.xpack.core.action.XPackUsageFeatureTransportAction;
+import org.elasticsearch.xpack.core.application.EnterpriseSearchFeatureSetUsage;
+
+import java.util.Map;
+
+public class EnterpriseSearchUsageTransportAction extends XPackUsageFeatureTransportAction {
+    private final SearchApplicationIndexService searchApplicationIndexService;
+
+    @Inject
+    public EnterpriseSearchUsageTransportAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        ActionFilters actionFilters,
+        IndexNameExpressionResolver indexNameExpressionResolver,
+        SearchApplicationIndexService searchApplicationIndexService
+    ) {
+        super(
+            XPackUsageFeatureAction.ENTERPRISE_SEARCH.name(),
+            transportService,
+            clusterService,
+            threadPool,
+            actionFilters,
+            indexNameExpressionResolver
+        );
+        this.searchApplicationIndexService = searchApplicationIndexService;
+    }
+
+    @Override
+    protected void masterOperation(
+        Task task,
+        XPackUsageRequest request,
+        ClusterState state,
+        ActionListener<XPackUsageFeatureResponse> listener
+    ) throws Exception {
+        searchApplicationIndexService.listSearchApplication("*", 0, 0, listener.delegateFailure((l, searchApplicationResult) -> {
+            l.onResponse(
+                new XPackUsageFeatureResponse(
+                    new EnterpriseSearchFeatureSetUsage(true, true, Map.of("count", searchApplicationResult.totalResults()))
```

Comment:
Is Enterprise Search always available and enabled? Shouldn't we check it like we did on `EnterpriseSearchInfoTransportAction`?

Reply:
looking at other modules and the response of `_xpack/usage` is depending on the license level,
`enabled` should be `XPackSettings.ENTERPRISE_SEARCH_ENABLED.get(settings)` and `available` should be `LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState)`.

example: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherUsageTransportAction.java

Reply:
✅ https://github.com/elastic/elasticsearch/pull/95306/commits/fb485618ba87f9e9c538dda60326e55109e96b24